great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py

```diff
@@ -0,0 +1,151 @@
+        Args:
+            name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            threshold: the ratio of values that must match a format string for it to be accepted
+            candidate_strings: a list of candidate date format strings that will REPLACE the default
+            additional_candidate_strings: a list of candidate date format strings that will SUPPLEMENT the default
+            data_context: DataContext
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        super().__init__(
+            name=name,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._metric_domain_kwargs = metric_domain_kwargs
+
+        self._threshold = threshold
+        if candidate_strings is not None:
+            self._candidate_strings = set(candidate_strings)
+        else:
+            self._candidate_strings = self.CANDIDATE_STRINGS
+
+        if additional_candidate_strings is not None:
+            self._candidate_strings.update(additional_candidate_strings)
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ):
+        """
+        Check the percentage of values matching each string, and return the best fit, or None if no
+        string exceeds the configured threshold.
+
+        :return: ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and
+        ptional details
+        """
+        validator: Validator = self.get_validator(
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        batch_ids: Optional[List[str]] = self.get_batch_ids(
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        if not batch_ids:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"Utilizing a {self.__class__.__name__} requires a non-empty list of batch identifiers."
+            )
+
+        nonnull_count = sum(
+            self.get_metrics(
+                batch_ids=batch_ids,
+                validator=validator,
+                metric_name="column_values.nonnull.count",
+                metric_domain_kwargs=self._metric_domain_kwargs,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
+            ).metric_values
+        )
+
+        format_string_success_ratios = dict()
```

Comment:
@NathanFarmer type hint please; also, we have a convention to use the shorthand for dictionary initialization, rather than the formal `dict()`.  Thanks.

Reply:
✅